src/gleam_erlang_ffi.erl

```diff
@@ -1,66 +1,21 @@
 -module(gleam_erlang_ffi).
 -export([
-    atom_from_dynamic/1, rescue/1, atom_from_string/1, get_line/1,
-    ensure_all_started/1, sleep/1, os_family/0, sleep_forever/0,
-    get_all_env/0, get_env/1, set_env/2, unset_env/1, demonitor/1,
-    new_selector/0, link/1, insert_selector_handler/3,
-    remove_selector_handler/2, select/1, select/2, trap_exits/1,
-    map_selector/2, merge_selector/2, flush_messages/0,
+    atom_from_string/1, sleep/1, sleep_forever/0, demonitor/1, new_selector/0,
+    link/1, insert_selector_handler/3, remove_selector_handler/2, select/1,
+    select/2, trap_exits/1, map_selector/2, merge_selector/2, flush_messages/0,
     priv_directory/1, connect_node/1, register_process/2, unregister_process/1,
-    process_named/1, identity/1, pid_from_dynamic/1, reference_from_dynamic/1,
-    port_from_dynamic/1, 'receive'/1, 'receive'/2
+    process_named/1, identity/1, 'receive'/1, 'receive'/2, new_name/0,
+    cast_down_message/1, cast_exit_reason/1
 ]).
 
--spec atom_from_string(binary()) -> {ok, atom()} | {error, atom_not_loaded}.
+-spec atom_from_string(binary()) -> {ok, atom()} | {error, nil}.
 atom_from_string(S) ->
     try {ok, binary_to_existing_atom(S)}
-    catch error:badarg -> {error, atom_not_loaded}
+    catch error:badarg -> {error, nil}
     end.
 
-atom_from_dynamic(Data) when is_atom(Data) ->
-    {ok, Data};
-atom_from_dynamic(Data) ->
-    {error, [{decode_error, <<"Atom">>, gleam@dynamic:classify(Data), []}]}.
-
-pid_from_dynamic(Data) when is_pid(Data) ->
-    {ok, Data};
-pid_from_dynamic(Data) ->
-    {error, [{decode_error, <<"Pid">>, gleam@dynamic:classify(Data), []}]}.
-
-reference_from_dynamic(Data) when is_reference(Data) ->
-    {ok, Data};
-reference_from_dynamic(Data) ->
-    {error, [{decode_error, <<"Reference">>, gleam@dynamic:classify(Data), []}]}.
-
-port_from_dynamic(Data) when is_port(Data) ->
-    {ok, Data};
-port_from_dynamic(Data) ->
-    {error, [{decode_error, <<"Port">>, gleam@dynamic:classify(Data), []}]}.
-
--spec get_line(io:prompt()) -> {ok, unicode:unicode_binary()} | {error, eof | no_data}.
-get_line(Prompt) ->
-    case io:get_line(Prompt) of
-        eof -> {error, eof};
-        {error, _} -> {error, no_data};
-        Data when is_binary(Data) -> {ok, Data};
-        Data when is_list(Data) -> {ok, unicode:characters_to_binary(Data)}
-    end.
-
-rescue(F) ->
-    try {ok, F()}
-    catch
-        throw:X -> {error, {thrown, X}};
-        error:X -> {error, {errored, X}};
-        exit:X -> {error, {exited, X}}
-    end.
-
-ensure_all_started(Application) ->
-    case application:ensure_all_started(Application) of
-        {ok, _} = Ok -> Ok;
-
-        {error, {ProblemApp, {"no such file or directory", _}}} ->
-            {error, {unknown_application, ProblemApp}}
-    end.
+new_name() ->
+    list_to_atom("name" ++ integer_to_list(erlang:unique_integer([positive]))).
```

Comment:
Let the programmer decide the name, so the process will be easy to find with inspection/debugging tools, but keep appending the unique number.